app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/ui/browser/BrowserFragment.java

```diff
@@ -0,0 +1,262 @@
+        forwardMenuItem.setEnabled(enabled);
+    }
+
+    @Override
+    public void setRefreshEnabled(boolean enabled) {
+        MenuItem reloadMenuItem = toolbar.getMenu().findItem(R.id.action_refresh);
+        reloadMenuItem.setEnabled(enabled);
+    }
+
+    @Override
+    public void showProgressBar() {
+        if(progressBar.getVisibility() == View.GONE) {
+            progressBar.setVisibility(View.VISIBLE);
+            progressBar.animate().alpha(1);
+        }
+    }
+
+    @Override
+    public void hideProgressBar() {
+        final WeakReference<ProgressBar> progressBarWeakReference = new WeakReference<>(progressBar);
+        progressBar.animate().setStartDelay(400).alpha(0).setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                super.onAnimationEnd(animation);
+                ProgressBar progressBar = progressBarWeakReference.get();
+                if(progressBar != null) {
+                    progressBar.setVisibility(View.GONE);
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onProgressChanged(int newProgress) {
+        progressBar.setProgress(newProgress);
+    }
+
+    private void initUI() {
+        initWebView(webView);
+        initToolbar(toolbar);
+        initSearchEditText(searchEditText);
+    }
+
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private void initWebView(WebView webView) {
+        WebSettings webSettings = webView.getSettings();
+        webSettings.setJavaScriptCanOpenWindowsAutomatically(false);
+        webSettings.setJavaScriptEnabled(true);
+        webView.setWebViewClient(new DDGWebViewClient(browserPresenter));
+        webView.setWebChromeClient(new DDGWebChromeClient(browserPresenter));
+    }
+
+    private void initToolbar(Toolbar toolbar) {
+        toolbar.inflateMenu(R.menu.browser);
+        toolbar.setOnMenuItemClickListener(new Toolbar.OnMenuItemClickListener() {
+            @Override
+            public boolean onMenuItemClick(MenuItem item) {
+                switch(item.getItemId()) {
+                    case R.id.action_go_back:
+                        browserPresenter.navigateHistoryBackward();
+                        return true;
+                    case R.id.action_go_forward:
+                        browserPresenter.navigateHistoryForward();
+                        return true;
+                    case R.id.action_refresh:
+                        browserPresenter.refreshCurrentPage();
+                        return true;
+                }
+                return false;
+            }
+        });
+    }
+
+    private void initSearchEditText(EditText searchEditText) {
+        searchEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if(actionId == EditorInfo.IME_ACTION_SEARCH) {
```

Comment:
It might be worth responding to the enter key too so that we play well with attached keyboards. You can capture this with`KeyEvent.ACTION_DOWN` and `KeyEvent.KEYCODE_ENTER` and you could wrapp that in a boolean function to keep things readable. Maybe something like this?
```
private boolean wasEnterPressed(KeyEvent event) {
     return event.getAction() == KeyEvent.ACTION_DOWN && event.getKeyCode() == KeyEvent.KEYCODE_ENTER;
 }
```